gateway/api_test.go

```diff
@@ -1073,8 +1072,20 @@ func (ts *Test) testHashFuncAndBAHelper(t *testing.T) {
 			Code:      200,
 		},
 		{
-			Method:    "GET",
-			Path:      "/tyk/keys/defaultuser?username=true&org_id=default",
+			Method: "GET",
+			Path:   "/tyk/keys/defaultuser?username=true&org_id=default",
+			BodyMatchFunc: func(resp []byte) bool {
+				keyResp := user.SessionState{}
+				err := json.Unmarshal(resp, &keyResp)
+				if err != nil {
```

Comment:
asser.Error can be used

Reply:
Correct, prefer `assert.NoError(t, err)`; an `if` clause is a logical branch that counts towards code coverage, assert checks are non-conditional and automatically count towards code coverage.

See for other if style checks below, not just for err.